### PR TITLE
UCP/WORKER: Rename ucp_ep_cfg_index_t to ucp_worker_cfg_index_t

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Preparation for adding Rkey configuration (similarly to ep configuration) which will allow protocol selection according to remote memory type/locality.

### CODE REVIEW

**Comment by hoopoepg** [comment]
```
[2020-07-11T01:03:57.124Z] /scrap/jenkins/workspace/ucx/build-test/ucx-1.10.0/_build/sub/../../src/ucp/core/ucp_ep.inl: In function ‘ucp_ep_config_t* ucp_ep_config(ucp_ep_h)’:
[2020-07-11T01:03:57.124Z] /scrap/jenkins/workspace/ucx/build-test/ucx-1.10.0/_build/sub/../../src/ucp/core/ucp_types.h:45:38: error: ‘UINT8_MAX’ was not declared in this scope
[2020-07-11T01:03:57.124Z]  #define UCP_WORKER_CFG_INDEX_NULL    UINT8_MAX
```
it seems missing include file

File: src/ucp/core/ucp_mm.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ enum {
 typedef struct ucp_rkey {
     /* cached values for the most recent endpoint configuration */
     struct {
-        ucp_ep_cfg_index_t        ep_cfg_index; /* EP configuration relevant for the cache */
+        ucp_worker_cfg_index_t    ep_cfg_index; /* EP configuration relevant for the cache */
```
Comment (brminich) [inline src/ucp/core/ucp_mm.h:53]: `worker_cfg_index` or just `cfg_index`?

Comment (yosefe) [reply src/ucp/core/ucp_mm.h:53]: worker_cfg_index could help in case there is other configuration which is not on the worker (maybe on the context)

Comment (brminich) [reply src/ucp/core/ucp_mm.h:53]: i was just proposing a name change

**Pull request merged**